src/Model/DICOMDirectorySearch.py

```diff
@@ -77,19 +65,34 @@ def get_dicom_structure(path, interrupt_flag, progress_callback):
                         new_study.study_description = dicom_file.get("StudyDescription")
                         new_study.add_series(new_series)
 
-                        existing_patient.add_study(new_study)
+                        new_patient = Patient(patient_id, dicom_file.PatientName)
+                        new_patient.add_study(new_study)
+
+                        dicom_structure.add_patient(new_patient)
                     else:
-                        existing_study = existing_patient.get_study(dicom_file.StudyInstanceUID)
-                        if not existing_study.has_series(dicom_file.SeriesInstanceUID):
+                        existing_patient = dicom_structure.get_patient(dicom_file.PatientID)
+                        if not existing_patient.has_study(dicom_file.StudyInstanceUID):
                             new_series = Series(dicom_file.SeriesInstanceUID)
                             new_series.series_description = dicom_file.get("SeriesDescription")
                             new_series.add_image(new_image)
 
-                            existing_study.add_series(new_series)
+                            new_study = Study(dicom_file.StudyInstanceUID)
+                            new_study.study_description = dicom_file.get("StudyDescription")
+                            new_study.add_series(new_series)
+
+                            existing_patient.add_study(new_study)
                         else:
-                            existing_series = existing_study.get_series(dicom_file.SeriesInstanceUID)
-                            if not existing_series.has_image(dicom_file.SOPInstanceUID):
-                                existing_series.add_image(new_image)
+                            existing_study = existing_patient.get_study(dicom_file.StudyInstanceUID)
+                            if not existing_study.has_series(dicom_file.SeriesInstanceUID):
+                                new_series = Series(dicom_file.SeriesInstanceUID)
+                                new_series.series_description = dicom_file.get("SeriesDescription")
+                                new_series.add_image(new_image)
+
+                                existing_study.add_series(new_series)
+                            else:
+                                existing_series = existing_study.get_series(dicom_file.SeriesInstanceUID)
```

Comment:
Line length exceeds the maximum 99 characters by 5 characters - is this negligible?